### PR TITLE
Use literal '/'s to build internal zip file paths

### PR DESCRIPTION
**Pull request by earhart**
Necessary in order to work on Windows

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by earhart** [comment]
I signed it!

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: rules_python/whl.py [hunk 1]
```diff
@@ -60,12 +61,12 @@ def metadata(self):
     with zipfile.ZipFile(self.path(), 'r') as whl:
       # first check for metadata.json
       try:
-        with whl.open(os.path.join(self._dist_info(), 'metadata.json')) as f:
+        with whl.open(posixpath.join(self._dist_info(), 'metadata.json')) as f:
```
Comment (duggelz) [inline rules_python/whl.py]: Is `posixpath` always available on Windows and other non-Posix platforms?  From the [zip specification](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) and the code for the `zipfile` module, it appears that literal forward slash '/' is the only thing that can be used here, so we could safely simplify this if we wanted to:

`with whl.open(self._dist_info() + '/metadata.json') as f`

**Comment by duggelz** [comment]
Jenkins: test this please

**Comment by earhart** [comment]
I ran into this on Windows, and tested the fix there. :-) I don’t know about other non-Posix platforms, but other than Windows, I don’t know any that are supported by Bazel.

Using a literal slash should also work fine; want me to make the change?

**Comment by duggelz** [comment]
Yes, please go ahead and change to literal slashes.

**Comment by earhart** [comment]
Done

**Comment by earhart** [comment]
FYI -- without write access to the repository, I can't actually merge the pull request myself; someone else will have to do that.  :-)

**Comment by duggelz** [comment]
Jenkins: test this please

**Comment by duggelz** [comment]
@mattmoor I think the CI is wedged, and I don't have permission to force a merge.

**Comment by mattmoor** [comment]
Test this please

**Comment by mattmoor** [comment]
Oh, it got renamed.  @davidstanke that's annoying :)

**Comment by mattmoor** [comment]
This is the protected branch feature, you should be able to mess with it under `Settings > Branches > master (edit button)`.

**Pull request merged**